webapp/pgconfig2.py

```diff
@@ -0,0 +1,200 @@
+"""Parses PostgreSQL config data from pickled files"""
+import os
+import logging
+import pandas as pd
+import pickle
+
+from webapp import config
+
+LOGGER = logging.getLogger(__name__)
+VERSIONS = ['15', '12']
+
+
+NEW_STRING = 'Configuration parameter added'
+REMOVED_STRING = 'Configuration parameter removed'
+
+
+def config_changes(vers1: int, vers2: int) -> pd.DataFrame:
+    """Find changes between `vers1` and `vers2`.
+
+    Parameters
+    ----------------
+    vers1 : int
+        Version number, e.g. 11 or 16
+
+    vers2 : int
+        Version number, e.g. 11 or 16
+
+    Returns
+    -----------------
+    changed : pd.DataFrame
+    """
+    if vers2 <= vers1:
+        raise ValueError('Version 1 must be lower (before) version 2.')
+
+    # Have to drop enumvals in comparison to avoid errors comparing using pandas
+    data1 = load_config_data(pg_version=vers1).drop(columns='enumvals')
+    data2 = load_config_data(pg_version=vers2).drop(columns='enumvals')
+
+    data2 = data2.add_suffix('2')
+
+    combined = pd.concat([data1, data2], axis=1)
+
+    combined['summary'] = combined.apply(classify_changes, axis=1)
+    columns = ['summary', 'vartype', 'vartype2',
+            'boot_val_display', 'boot_val_display2']
+    changed = combined[combined['summary'] != ''][columns]
+
+    return changed
+
+
+def load_config_data(pg_version: int) -> pd.DataFrame:
+    """Loads the pickled config data for `pg_version` into DataFrame with the
+    config name as the index.
+
+    Returns empty DataFrame on file read error.
+
+    Parameters
+    ----------------------
+    pg_version : int
+
+    Returns
+    ----------------------
+    df : pd.DataFrame
+    """ 
+    base_path = os.path.dirname(os.path.realpath(__file__))
+    filename = os.path.join(base_path, 'config', f'pg{pg_version}.pkl')
+
+    try:
+        with open(filename, 'rb') as data_file:
```

Comment:
This was dismissed b/c code intermediately preceding this code ensures only a select list of values are accepted.

```python
 # Checking user input against configured versions to avoid security concerns
    if str(pg_version) not in VERSIONS:
        raise ValueError(f'Invalid Postgres version.  Options are {VERSIONS}')
```